### DISCUSSION TITLE
Using a custom translation

### DISCUSSION THREAD

**Discussion by vb-03** [pretix/pretix#2946] [Q&A] [answered]
Hi,
I've translated some strings in the translation platform in pt_PT and downloaded the file to start using it in my deployment.

I've followed the instructions in https://docs.pretix.eu/en/latest/admin/config.html#translations :
```
[languages]
enabled=en,pt-pt,pt
#allow_incubating=
path=/etc/pretix/customtranslations #(to make sure that docker can access it). 
```

Then I proceeded to put the files in the indicated structure:
/etc/pretix/customtranslations/pt_PT/LC_MESSAGES/django.po and 
/etc/pretix/customtranslations/pt/LC_MESSAGES/django.po, (wasn't sure if it was necessary, also modified the file header accordingly).

However, it keeps using the default translation. What am I missing here?

**Comment by raphaelm** [comment, marked as answer]
You need to compile the .po file to a .mo file using the `msgfmt` command

**Reply by vb-03** [reply]
And then use the generated "messages.mo" into the directory as it is?

**Reply by raphaelm** [reply]
Yes, that should work

**Reply by vb-03** [reply]
Unfortunately, it's not working :(
I've tried putting both files in the folder, even renaming messages.mo to django.po as in the docs but it's not working...
Does it need any specific naming?
I also changed my config like this:

```
[languages]
enabled=en,pt-pt,pt
allow_incubating=pt-pt
path=/etc/pretix/customtranslations
```

**Reply by raphaelm** [reply]
Yes, it needs special naming and folders. It needs to look like the src/pretix/locale folder in this repo.

**Reply by vb-03** [reply]
The files in the repo are uncompiled.... 
Should I gave the compiled files the "django.po" naming?
I also didn't have yet the djangojs.po file, and even tho the "test string" I'm using is not in there, could that be the issue?

**Reply by raphaelm** [reply]
no, the compiled file needs to be django.mo. You don't need the djangojs one. More info on translations in Django are here: https://docs.djangoproject.com/en/4.1/topics/i18n/translation/#localization-how-to-create-language-files

note that you might need to restart the pretix process after you placed the files

**Reply by vb-03** [reply]
Worked like a charm! Thanks!
